src/main/java/ro/msg/learning/shop/mappers/ProductMapper.java

```diff
@@ -0,0 +1,43 @@
+package ro.msg.learning.shop.mappers;
+
+import org.springframework.stereotype.Component;
+import ro.msg.learning.shop.dtos.ProductDTO;
+import ro.msg.learning.shop.entitites.Product;
+import ro.msg.learning.shop.entitites.ProductCategory;
+import ro.msg.learning.shop.entitites.Supplier;
+
+@Component
+public class ProductMapper {
+
+	public static ProductDTO toDTO(Product product) {
+		return new ProductDTO(
+			product.getId(),
+			product.getName(),
+			product.getDescription(),
+			product.getPrice(),
+			product.getWeight(),
+			product.getImageUrl(),
+			product.getCategory().getId(),
+			product.getSupplier().getId()
+		);
+	}
+
+	public static Product toEntity(ProductDTO dto) {
+			var productCategory = new ProductCategory();
```

Comment:
This code has a weird indentation. Please check your Intellij settings: 
![image](https://github.com/ro-msg-spring-training/online-shop-ghetze/assets/127387393/62a9f7c9-9eed-494e-a4f5-fc358e80301c)
When the settings are correct, please make sure that al of your files are formatted correctly.

Reply:
Done.